x/distribution/keeper/keeper_test.go

```diff
@@ -0,0 +1,129 @@
+			},
+			GetModuleAddressFunc: func(name string) sdk.AccAddress {
+				return authtypes.NewModuleAddress(name)
+			},
+		}
+		bk = &mock.BankKeeperMock{
+			GetAllBalancesFunc: func(ctx sdk.Context, addr sdk.AccAddress) sdk.Coins {
+				return accBalances[addr.String()]
+			},
+			SendCoinsFromModuleToModuleFunc: func(ctx sdk.Context, senderModule, recipientModule string, amt sdk.Coins) error {
+				senderModule = authtypes.NewModuleAddress(senderModule).String()
+				recipientModule = authtypes.NewModuleAddress(recipientModule).String()
+
+				accBalances[senderModule] = accBalances[senderModule].Sub(amt)
+				accBalances[recipientModule] = accBalances[recipientModule].Add(amt...)
+
+				return nil
+			},
+			SendCoinsFromModuleToAccountFunc: func(ctx sdk.Context, senderModule string, recipientAddr sdk.AccAddress, amt sdk.Coins) error {
+				senderModule = authtypes.NewModuleAddress(senderModule).String()
+
+				accBalances[senderModule] = accBalances[senderModule].Sub(amt)
+				accBalances[recipientAddr.String()] = accBalances[recipientAddr.String()].Add(amt...)
+
+				return nil
+			},
+			BurnCoinsFunc: func(ctx sdk.Context, name string, amt sdk.Coins) error {
+				acc := authtypes.NewModuleAddress(name).String()
+				accBalances[acc] = accBalances[acc].Sub(amt)
+
+				return nil
+			},
+			MintCoinsFunc: func(ctx sdk.Context, name string, amt sdk.Coins) error {
+				acc := authtypes.NewModuleAddress(name).String()
+				accBalances[acc] = accBalances[acc].Add(amt...)
+
+				return nil
+			},
+		}
+		sk := &mock.StakingKeeperMock{
+			ValidatorByConsAddrFunc: func(ctx sdk.Context, addr sdk.ConsAddress) stakingtypes.ValidatorI {
+				seed := []byte("key")
+				consKey := ed25519.GenPrivKeyFromSecret(seed).PubKey()
+				pk := secp256k1.GenPrivKeyFromSecret(seed)
+				valAddr := sdk.ValAddress(pk.PubKey().Address().Bytes())
+				return funcs.Must(stakingtypes.NewValidator(valAddr, consKey, stakingtypes.Description{}))
+			},
+		}
+
+		distriK := distribution.NewKeeper(encCfg.Codec, sdk.NewKVStoreKey(distributiontypes.StoreKey), subspace, ak, bk, sk, authtypes.FeeCollectorName, map[string]bool{})
+		k = keeper.NewKeeper(distriK, ak, bk, sk, authtypes.FeeCollectorName)
+		k.SetFeePool(ctx, distributiontypes.FeePool{CommunityPool: sdk.DecCoins{}})
+		k.SetParams(ctx, distributiontypes.DefaultParams())
+	}).
+		When("allocate token", func() {
+			k.AllocateTokens(ctx, 0, 1, sdk.ConsAddress{}, nil)
+		}).
+		Then("allocate to community pool and burn the rest", func(t *testing.T) {
+			assert.Len(t, bk.BurnCoinsCalls(), 1)
+
+			feeBurnedType := proto.MessageName(&types.FeeBurnedEvent{})
+			assert.Len(t, slices.Filter(ctx.EventManager().Events(), func(e sdk.Event) bool {
+				return e.Type == feeBurnedType
+			}), 1)
+
+			expectedBurnedFee := sdk.NewCoins(slices.Map(expectedBurnedFee(ctx, k, fee), types.WithBurnedPrefix)...)
+			assert.Equal(t, expectedBurnedFee, accBalances[types.ZeroAddress.String()])
```

Comment:
are we testing that the correct amount gets allocated to the community pool?

Reply:
added